circuitpython_nrf24l01/rf24.py

```diff
@@ -123,29 +136,34 @@ def __init__(
     def __enter__(self):
         self._ce_pin.value = False
         self._config |= 2
-        self._reg_write(CONFIGURE, self._config)
+        if self._aa & 1:
+            self._open_pipes |= 1
+            self._reg_write_bytes(_RX_ADDR_P0, self._tx_address)
+        self._reg_write(_CONFIG, self._config)
         # time.sleep(0.00015)  # let the rest of this function be the delay
-        self._reg_write(RF_PA_RATE, self._rf_setup)
-        self._reg_write(OPEN_PIPES, self._open_pipes)
-        self._reg_write(DYN_PL_LEN, self._dyn_pl)
-        self._reg_write(AUTO_ACK, self._aa)
-        self._reg_write(TX_FEATURE, self._features)
-        self._reg_write(SETUP_RETR, self._retry_setup)
+        self._reg_write(_RF_SETUP, self._rf_setup)
+        self._reg_write(_EN_RX, self._open_pipes)
+        self._reg_write(_DYN_PL, self._dyn_pl)
+        self._reg_write(_EN_AA, self._aa)
+        self._reg_write(_FEATURE, self._features)
+        self._reg_write(_SETUP_RETR, self._retry_setup)
         for i, addr in enumerate(self._pipes):
-            if i < 2:
-                self._reg_write_bytes(RX_ADDR_P0 + i, addr)
-            else:
-                self._reg_write(RX_ADDR_P0 + i, addr)
             self.set_payload_length(self._pl_len[i], i)
-        self._reg_write_bytes(TX_ADDRESS, self._tx_address)
+            if not i:  # skip pipe 0 RX address because we're going into TX mode
+                continue
+            if i == 1:
+                self._reg_write_bytes(_RX_ADDR_P0 + i, addr)
```

Comment:
```suggestion
            if not i and self._aa & 1:
                # skip pipe 0 RX address because we're going into TX mode
                continue
            elif i < 2:
                self._reg_write_bytes(_RX_ADDR_P0 + i, addr)
```
Should also add consideration for cached config register and `_is_p0_rx` states (for `__enter__()` as a whole, not just here).